src/main/java/org/broadinstitute/hellbender/tools/exome/GetBayesianHetCoverage.java

```diff
@@ -0,0 +1,300 @@
+            optional = true
+    )
+    protected double maximumAbnormalFraction = 0.8;
+
+    @Argument(
+            doc = "Estimated maximum copy number for either homologs in the abnormal portion of the sample (for " +
+                    "building the allele fraction prior; to be used for tumor-only cases)",
+            fullName = MAXIMUM_COPY_NUMBER_FULL_NAME,
+            shortName = MAXIMUM_COPY_NUMBER_SHORT_NAME,
+            optional = true
+    )
+    protected int maximumCopyNumber = 2;
+
+    @Argument(
+            doc = "Integration quadrature order; a good choice is the typical read depth (to be used for tumor-only cases)",
+            fullName = QUADRATURE_ORDER_FULL_NAME,
+            shortName = QUADRATURE_ORDER_SHORT_NAME,
+            optional = true
+    )
+    protected int quadratureOrder = 200;
+
+    @Argument(
+            doc = "(Experimental) error probability adjustment factor (could be any positive value)",
+            fullName = ERROR_PROBABILITY_ADJUSTMENT_FACTOR_FULL_NAME,
+            shortName = ERROR_PROBABILITY_ADJUSTMENT_FACTOR_SHORT_NAME,
+            optional = true
+    )
+    protected double errorProbabilityAdjustmentFactor = 1.0;
+
+    private enum HetCoverageJobType {
+        NORMAL_ONLY, TUMOR_ONLY, NORMAL_TUMOR, NONE
+    }
+
+    @Override
+    protected Object doWork() {
+
+        /* check for illegal arguments */
+        if (normalBamFile == null && tumorBamFile == null) {
+            throw new UserException.CommandLineException("At least one of normal and tumor BAM files must be provided. Stopping.");
+        }
+        if (normalBamFile != null && normalHetOutputFile == null) {
+            throw new UserException.CommandLineException("If the normal BAM file is provided, the normal Het output file must also be provided. Stopping.");
+        }
+        if (tumorBamFile != null && tumorHetOutputFile == null) {
+            throw new UserException.CommandLineException("If the tumor BAM file is provided, the tumor Het output file must also be provided. Stopping.");
+        }
+
+        /* determine the job type */
+        HetCoverageJobType jobType = HetCoverageJobType.NONE;
+        if (normalBamFile != null && tumorBamFile != null) {
+            jobType = HetCoverageJobType.NORMAL_TUMOR;
+            logger.info("NORMAL_TUMOR mode selected.");
+        }
+        if (normalBamFile != null && tumorBamFile == null) {
+            jobType = HetCoverageJobType.NORMAL_ONLY;
+            logger.info("NORMAL_ONLY mode selected.");
+        }
+        if (normalBamFile == null && tumorBamFile != null) {
+            jobType = HetCoverageJobType.TUMOR_ONLY;
+            logger.info("TUMOR_ONLY mode selected.");
+        }
+
+        BayesianHetPulldownCalculator normalHetPulldownCalculator, tumorHetPulldownCalculator;
+        Pulldown normalHetPulldown, tumorHetPulldown;
+
+        /* run the job */
+        switch (jobType) {
+
+            case NORMAL_ONLY:
+
```

Comment:
In Java, switch cases code normally is not that long. Please, extract private methods out of each case but trying to share common code.